ds-image-analysis-ws/src/main/java/dk/kb/api/webservice/ApiServiceImpl.java

```
@@ -0,0 +1,186 @@
+                if ((hash1.compareTo(BigInteger.ZERO) == -1) || (hash2.compareTo(BigInteger.ZERO) == -1)) {
+                    reply.setMessage("Both hash values has to be positive");
+                    return reply;
+                }
+
+                int distanceBits = 0;
+                // Increase distanceBits by going through the xor'ed number
+                BigInteger x = hash1.xor(hash2);
+                while (x.compareTo(BigInteger.ZERO) == 1)
+                {
+                    // Similar to distanceBits += x & 1
+                    if (x.and(BigInteger.ONE).compareTo(BigInteger.ZERO) == 1) {
+                        distanceBits++;
+                    }
+                    x = x.shiftRight(1);
+                }
+                reply.setMessage(Integer.toString(distanceBits));
+            } catch (NumberFormatException nfe) {
+                reply.setMessage("Both values has to be BigInteger separated by ; character");
+                return reply;
+            }
+        }
+        else {
+            reply.setMessage("The two hash values has to be divided by ; character");
+        }
+        return reply;
+    }
+
+    /**
+     * Returns the discrete hash values of the image in imgURL
+     * @param imgURL The path to the image given to the webservice.
+     * @return The discrete hash values
+     */
+    @Override
+    public HashReplyDto getImageDHash(String imgURL) {
+        HashReplyDto reply = new HashReplyDto();
+        try {
+            File imgHash = getFile(imgURL);
+
+            hashValue(replyBuilder, imgHash, DifferenceHash.Precision.Simple);
+            hashValue(replyBuilder, imgHash, DifferenceHash.Precision.Double);
+            hashValue(replyBuilder,imgHash, DifferenceHash.Precision.Triple);
+            reply.setMessage(replyBuilder.toString());
+
+            Files.deleteIfExists(Paths.get(filePath));
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```

Please send an error back to the caller instead of (from the caller's perspective) silently fail.
Also please use `log.error("Description of what went wrong", e)` instead of `e.printStackTrace()`.

---

Done